test/inductor/mock_cache.py

```diff
@@ -1,127 +1,114 @@
+        return self
+
+    def reset(self) -> None:
         self.num_put = 0
         self.num_get_hit = 0
         self.num_get_miss = 0
 
-    def report(self):
-        print(
-            "".join(
-                [
-                    f"{self._name} cache: ",
-                    f"init: {self.num_init}, ",
-                    f"puts: {self.num_put}, ",
-                    f"misses: {self.num_get_miss}, ",
-                    f"hits: {self.num_get_hit}, ",
-                ]
-            ),
-            file=sys.stderr,
+    def __str__(self) -> str:
+        return "".join(
+            (
+                f"puts: {self.num_put}, ",
+                f"misses: {self.num_get_miss}, ",
+                f"hits: {self.num_get_hit}, ",
+            )
         )
 
 
-class _MockLocalAutotuneCacheBackend(RemoteCacheBackend):
-    _state = _MockCacheState("Local")
+# The cache states are thread-local so if we're running multiple tests at once
+# they won't cross contaminate. However - it needs to be "global" because we
+# allow code to create new cache clients which refer to the same cache (because
+# it's a remote cache).
 
-    def __init__(self):
-        state = self._state
-        state.num_init += 1
 
-    @override
-    def get(self, key: str) -> Optional[bytes]:
-        assert isinstance(key, str)
+class _GlobalStats(Stats, threading.local):
+    def __init__(self) -> None:
+        self.autotune = Stats()
+        self.fx_graph = Stats()
+        self.triton = Stats()
 
-        state = self._state
-        if key in state._cache:
-            state.num_get_hit += 1
-            return state._cache[key]
-        else:
-            state.num_get_miss += 1
+    def reset(self) -> None:
+        self.autotune.reset()
+        self.fx_graph.reset()
+        self.triton.reset()
 
-    @override
-    def put(self, key: str, data: bytes) -> None:
-        assert isinstance(key, str)
-        assert isinstance(data, bytes)
+    def update(self, name: str, delta: Stats) -> None:
+        stat = getattr(self, name)
+        stat += delta
 
-        state = self._state
-        state.num_put += 1
-        state._cache[key] = data
+    def report(self):
+        print("Cache Stats:", file=sys.stderr)
+        print(f"  autotune: {self.autotune}", file=sys.stderr)
+        print(f"  fx_graph: {self.fx_graph}", file=sys.stderr)
+        print(f"  triton:   {self.triton}", file=sys.stderr)
 
 
-class _MockRedisRemoteCache:
-    _state = _MockCacheState("Redis")
+global_stats = _GlobalStats()
 
-    def __init__(self, *args, **kwargs):
-        state = self._state
-        state.num_init += 1
 
-    def get(self, key: Union[bytes, str]) -> Optional[Union[bytes, str, int, float]]:
-        assert isinstance(key, (bytes, str))
+class MockBackend(RemoteCacheBackend[Any]):
+    def __init__(self, name: str, cache: Dict[str, object]) -> None:
+        print(f"*** MockBackend.__init__({name})",file=sys.stderr)
```

Comment:
Missing space after the coma. 
There are few other of these.

Reply:
This is just debug cruft to try to figure out why some tests are failing in CI.
This is still a work in progress. Please don't waste energy reviewing it.

Reply:
You can cherry-pick 2f44196c203d4c7b853118d0a08121d174d006e0

Reply:
>  This is still a work in progress. Please don't waste energy reviewing it.

Oh I supposed we used the DRAFT status of the Github PR when not ready for review. But probably with the export is different.

Reply:
Let me know when it is done I will take the last commit to check what is the effect on 2.4.0 branch on the CI and its compatibility.